examples/trade-search.py

```diff
@@ -1,42 +1,31 @@
 import os
-import sys
-from dotenv import load_dotenv # type: ignore
-from sayari.client import Sayari
 from sayari.environment import SayariEnvironment
+from env_loader import load_env_vars_and_authenticate
 
-# NOTE: To connect you must provide your client ID and client secret. To avoid accidentally checking these into git,
-# it is recommended to use ENV variables
-# load ENV file if ENV vars are not set
-if os.getenv('CLIENT_ID') is None or os.getenv('CLIENT_SECRET') is None:
-    load_dotenv()
 
+def main():
+    """
+    Main function to search for shipments, suppliers, and buyers related to 'microcenter'.
+    Default environment is set to PRODUCTION unless BASE_URL is set.
+    """
+    # Set environment, default to PRODUCTION unless BASE_URL is set
+    base_url = os.getenv("BASE_URL", SayariEnvironment.PRODUCTION)
+    env = SayariEnvironment(base_url)
 
-client_id = os.getenv('CLIENT_ID')
-client_secret = os.getenv('CLIENT_SECRET')
-if client_id is None or client_secret is None:
-    print("The CLIENT_ID and CLIENT_SECRET environment variables are required to run this example.")
-    sys.exit(1)
+    # Load environment variables and authenticate
+    client = load_env_vars_and_authenticate(env)
 
-# Set ENV if provided (next 3 lines can be omitted if running in prod, as well as the env arg when creating the client.)
-env = SayariEnvironment.PRODUCTION
-if os.getenv('BASE_URL') is not None:
-    env = SayariEnvironment(os.getenv('BASE_URL'))
+    # Define search queries and corresponding entity names
+    queries = [
+        (client.trade.search_shipments(q="microcenter"), "shipments"),
+        (client.trade.search_suppliers(q="microcenter"), "suppliers"),
+        (client.trade.search_buyers(q="microcenter"), "buyers"),
+    ]
 
-# Create a client that is authed against the API
-client = Sayari(
-    client_id=client_id,
-    client_secret=client_secret,
-    environment=env
-)
+    # Search and print results
+    for entity, entity_name in queries:
```

Comment:
While this is more concise, I don't think it is as useful to someone learning our tool. That said, if we are going to keep this, lets change `entity_name` to `search_type` or something like that since I don't think the current format is accurate

Reply:
Accepted! Changed to search_type. I have also added some more inline documentation explaining how the query has been setup for someone learning our tool.